### ISSUE TITLE
aws_datasync_task accepts a CloudWatch log group but does not enable logging

### ISSUE THREAD

**Issue by tdmalone** [hashicorp/terraform-provider-aws#14666]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
Terraform v0.12.26
+ provider.aws v2.67.0
```

I'm a little out of date but I have verified from [the source](https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_datasync_task.go) that the same issue would exist as the ability to set the `LogLevel` option doesn't exist and the AWS default is `OFF`.

### Affected Resource(s)

* [aws_datasync_task](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/datasync_task)

### Terraform Configuration Files

```hcl
resource "aws_datasync_task" "main" {
  name = "my-task"

  source_location_arn      = aws_datasync_location_nfs.source.arn
  destination_location_arn = aws_datasync_location_efs.destination.arn

  cloudwatch_log_group_arn = replace(aws_cloudwatch_log_group.main.arn, "/:\\*$/", "") # NOTE: The :* suffix fix isn't needed beyond v3.0.0
}

resource "aws_cloudwatch_log_group" "main" {
  name = "my-task"
}
```

### Expected Behavior

Datasync task is created with logging enabled to the supplied CloudWatch log group.

### Actual Behavior

Datasync task is created, but logging is disabled:

![image](https://user-images.githubusercontent.com/900469/90297460-e3d2de00-ded1-11ea-8683-d275fe773ed2.png)

The CloudWatch log group ARN _is_ saved, but user must manually enable logging via the CLI (or AWS console) afterwards, eg.:

```bash
aws datasync update-task --task-arn arn:aws:datasync:REGION:ACCOUNT:task/task-TASK-ID --options LogLevel=OFF|BASIC|TRANSFER
```

### Steps to Reproduce

1. `terraform apply`
2. Check log settings on created task.

### Important Factoids

This is technically a request for supporting the LogLevel option, and could be implemented as follows:

```
  options {
    # Suggested configuration, not implemented.
    log_level = "BASIC" # Potential values: OFF, BASIC, TRANSFER
  }
```

I've logged it as a bug rather than a feature as it was unexpected behaviour to be able to provide a `cloudwatch_log_group_arn` but have logging turned `OFF` and not be able to be configured via Terraform. Apologies if this should have been a feature instead, I think it could go both ways really!

### References

Other `aws_datasync_task` feature requests:

* #10995 (scheduling)
* #11153 (filtering)

**Comment by tdmalone** [comment]
Workaround:

```hcl
resource "aws_datasync_task" "main" {
  ...

  # Sets the task LogLevel: OFF, BASIC or TRANSFER (default is OFF)
  # TODO: Remove this workaround when support is in Terraform, see terraform-providers/terraform-provider-aws#14666
  provisioner "local-exec" {
    command = "aws datasync update-task --task-arn ${self.arn} --options LogLevel=TRANSFER"
  }
}
```

**Comment by edlevin6612** [comment]
Thank you for the workaround, I also had to do something similar for transfer mode:

```
    # Sets the task TransferMode: ALL or CHANGED (default is CHANGED)
    # TODO: Remove this workaround when support is in Terraform
    provisioner "local-exec" {
        command = "aws datasync update-task --task-arn ${self.arn} --options TransferMode=ALL"
    }
```

**Comment by mattdrees** [comment]
This seems to have been duplicated by #18933 and implemented in  #18932. Probably ok to close this issue now.

**Comment by tdmalone** [comment]
Thanks for the heads up @mattdrees! Closing.
